examples/reshare/src/dkg/actor.rs

```diff
@@ -0,0 +1,397 @@
+                    .choose_multiple(
+                        &mut rng,
+                        self.num_participants_per_epoch - inactive_participants.len(),
+                    )
+                    .cloned()
+                    .collect::<Vec<_>>();
+                inactive_participants.extend_from_slice(dealer_players.as_slice());
+            } else if inactive_participants.len() > self.num_participants_per_epoch {
+                // Truncate the number of players if there are too many.
+                inactive_participants.truncate(self.num_participants_per_epoch);
+            }
+
+            // special case: If the starting epoch has already passed, we set the dealers for the current epoch
+            // as the first epoch's players, and randomly select a new set of players for the next epoch as
+            // usual.
+            if initial_epoch == 1 {
+                active_participants = inactive_participants.clone();
+                inactive_participants = all_participants
+                    .iter()
+                    .cloned()
+                    .choose_multiple(&mut rng, self.num_participants_per_epoch);
+            }
+        } else {
+            // If we're starting from a later epoch, we need to pseudorandomly select both the dealers
+            // and players for the current epoch, based on the epoch number as a seed.
+            let mut last_epoch_rng = StdRng::seed_from_u64(initial_epoch - 1);
+            active_participants = all_participants
+                .iter()
+                .cloned()
+                .choose_multiple(&mut last_epoch_rng, self.num_participants_per_epoch);
+            inactive_participants = all_participants
+                .iter()
+                .cloned()
+                .choose_multiple(&mut rng, self.num_participants_per_epoch);
+        }
+
+        // Initialize the DKG manager for the first round.
+        let mut manager = DkgManager::init(
+            &mut self.context,
+            initial_epoch,
+            initial_public,
+            initial_share,
+            &mut self.signer,
+            Set::from_iter(active_participants),
+            Set::from_iter(inactive_participants),
+            &mut dkg_mux,
+            &mut self.round_metadata,
+        )
+        .await;
+
+        while let Some(message) = self.mailbox.next().await {
+            match message {
+                Message::Act { response } => {
+                    let outcome = manager.take_deal_outcome();
+
+                    if let Some(ref outcome) = outcome {
+                        info!(
+                            n_acks = outcome.acks.len(),
+                            n_reveals = outcome.reveals.len(),
+                            "including reshare outcome in proposed block"
+                        );
+                    }
+
+                    let _ = response.send(outcome);
+                }
+                Message::Finalized { block } => {
```

Comment:
nit: add a comment that this can be used to mark a deal outcome as "sent" (so we don't need to keep re-proposing?)